packages/core/src/components/forms/numericStepper.tsx

```diff
@@ -0,0 +1,319 @@
+        }
+        if (majorStepSize <= 0) {
+            throw new Error(Errors.NUMERIC_STEPPER_MAJOR_STEP_SIZE_NON_POSITIVE);
+        }
+        if (stepSize <= 0) {
+            throw new Error(Errors.NUMERIC_STEPPER_STEP_SIZE_NON_POSITIVE);
+        }
+        if (minorStepSize > stepSize) {
+            throw new Error(Errors.NUMERIC_STEPPER_MINOR_STEP_SIZE_BOUND);
+        }
+        if (majorStepSize < stepSize) {
+            throw new Error(Errors.NUMERIC_STEPPER_MAJOR_STEP_SIZE_BOUND);
+        }
+    }
+
+    private renderInputGroup() {
+        return (
+            <InputGroup
+                {...removeNonHTMLProps(this.props)}
+                intent={this.props.intent}
+                inputRef={this.inputRef}
+                key="input-group"
+                leftIconName={this.props.leftIconName}
+                onBlur={this.handleInputBlur}
+                onChange={this.handleInputChange}
+                onKeyDown={this.handleKeyDown}
+                value={this.state.value}
+            />
+        );
+    }
+
+    private renderButton(key: string, iconName: string, onClick: React.MouseEventHandler<HTMLElement>) {
+        return (
+            <Button
+                disabled={this.props.disabled || this.props.readOnly}
+                iconName={iconName}
+                intent={this.props.intent}
+                key={key}
+                onClick={onClick}
+            />
+        );
+    }
+
+    private inputRef = (input: HTMLInputElement) => {
+        this.inputElement = input;
+    }
+
+    private handleDecrementButtonClick = (e: React.MouseEvent<HTMLInputElement>) => {
+        this.updateValue(-1, e);
+    }
+
+    private handleIncrementButtonClick = (e: React.MouseEvent<HTMLInputElement>) => {
+        this.updateValue(+1, e);
+    }
+
+    private handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+        if (this.props.disabled || this.props.readOnly) {
+            return;
+        }
+
+        const { keyCode } = e;
+
+        let direction: number;
+
+        if (keyCode === Keys.ARROW_UP) {
+            direction = 1;
+        } else if (keyCode === Keys.ARROW_DOWN) {
+            direction = -1;
+        } else {
+            return;
+        }
+
+        // we'd like to select the field contents after running the code in this
+        // onKeyDown handler, as a UX nicety. without e.preventDefault, some
+        // hotkeys (e.g. shift + up/down, alt + up/down) will clear the selection,
+        // resulting in an inconsistent or unintuitive experience.
+        e.preventDefault();
+
+        this.updateValue(direction, e);
+    }
+
+    private handleInputBlur = () => {
+        const currValue = this.state.value;
+        const nextValue = (currValue.length > 0) ? this.getAdjustedValue(currValue, /* delta */ 0) : "";
+        this.setState({ value: nextValue });
+
+        this.invokeOnChangeCallback(nextValue);
```

Comment:
I'd also like to add a prop for an `onSubmit` callback (triggered on <kbd>Enter</kbd> and `onblur`) that can accept the current `value`, evaluate it somehow (e.g. as a mathematical expression), and feed the result back into the component as a new prop if desired.